auslib/web/admin/views/permissions.py

```diff
@@ -66,67 +65,79 @@ def get(self, username, permission):
         try:
             perm = dbo.permissions.getUserPermissions(username)[permission]
         except KeyError:
-            return Response(status=404)
+            return problem(404, "Not Found", "Requested user permission"
+                                             " %s not found for %s" % (permission, username))
         return jsonify(perm)
 
     @requirelogin
     def _put(self, username, permission, changed_by, transaction):
         try:
             if dbo.permissions.getUserPermissions(username, transaction).get(permission):
-                form = ExistingPermissionForm()
-                if not form.validate():
-                    self.log.warning("Bad input: %s", form.errors)
-                    return Response(status=400, response=json.dumps(form.errors))
-                dbo.permissions.update(where={"username": username, "permission": permission}, what={"options": form.options.data},
-                                       changed_by=changed_by, old_data_version=form.data_version.data, transaction=transaction)
-                new_data_version = dbo.permissions.getPermission(username=username, permission=permission, transaction=transaction)['data_version']
+                # Existing Permission
+                if not connexion.request.json.get("data_version"):
+                    return problem(400, "Bad Request", "'data_version' is missing from request body")
+
+                options_dict = None if not connexion.request.json.get("options") else json.loads(
```

Comment:
Can you split this (and the subsequent similar code) into multiple lines? Something like:
```
options_dict = None
if connexion.request.json.get("options"):
    options_dict = json.loads(...)
```

...is much easier to read.